client/injection/apiextensions/reconciler/apiextensions/v1/customresourcedefinition/reconciler.go

```diff
@@ -254,20 +258,22 @@ func (r *reconcilerImpl) Reconcile(ctx context.Context, key string) error {
 		reconcileEvent = rof.ObserveFinalizeKind(ctx, resource)
 	}
 
-	// Synchronize the status.
-	if equality.Semantic.DeepEqual(original.Status, resource.Status) {
-		// If we didn't change anything then don't call updateStatus.
-		// This is important because the copy we loaded from the injectionInformer's
-		// cache may be stale and we don't want to overwrite a prior update
-		// to status with this stale state.
-	} else if !isLeader {
-		logger.Warn("Saw status changes when we aren't the leader!")
-		// TODO: Consider logging the diff at Debug?
-	} else if err = r.updateStatus(original, resource); err != nil {
-		logger.Warnw("Failed to update resource status", zap.Error(err))
-		r.Recorder.Eventf(resource, corev1.EventTypeWarning, "UpdateFailed",
-			"Failed to update status for %q: %v", resource.Name, err)
-		return err
+	if !r.skipStatusUpdates {
```

Comment:
I wonder if this would read better as:

```
switch {
case !r.skipStatusUpdates:
   // The reconciler has explicitly opted-out of updates
case !isLeader:
   // This reconciler is on an observing replica. Only the leader-elected pod makes updates.
case equality.Semantic.DeepEqual(original.Status, resource.Status):
   // No write-back to make.
default:
  if err = r.updateStatus(original, resource); err != nil {
  }
}
```